pkg/cmdutil/repo_override.go

```diff
@@ -51,20 +51,49 @@ func EnableRepoOverride(cmd *cobra.Command, f *Factory) {
 		if err := executeParentHooks(cmd, args); err != nil {
 			return err
 		}
-		repoOverride, _ := cmd.Flags().GetString("repo")
-		f.BaseRepo = OverrideBaseRepoFunc(f, repoOverride)
+
+		// First, get the value of the flag.
+		// We can ignore errors here because the flag is guaranteed to exist and be string.
+		userProvidedRepo, _ := cmd.Flags().GetString("repo")
+		if userProvidedRepo == "" {
+			// If there was no flag set, then check the GH_REPO environment variable,
+			// which has lower precedence.
+			repoOverride, ok := os.LookupEnv("GH_REPO")
```

Comment:
Just going to say this is more elegant of a solution than I was considering when we initially discussed this.  I hadn't considered the utility logic setting up the persistent flag.

This helps keep the concerns in 1 place, so hopefully easier to grok and manage.